M1/practica/index.html

```diff
@@ -2,107 +2,87 @@
 <html lang="en">
 <head>
     <meta charset="UTF-8">
-    <link rel='stylesheet' href='bower_components/bootstrap/dist/css/bootstrap.css'>
-    <script type='text/javascript' src="bower_components/jquery/dist/jquery.js"></script>
-    <title>Curso web</title>
-    <script type='text/javascript'>
-        var personas = [
-            {"nombre": "Alejandro", "apellido": "Marty", "edad": 24},
-            {"nombre": "Gasti", "apellido": "Pisera", "edad": 25},
-            {"nombre": "Chubi", "apellido": "Neitor", "edad": 45}
-        ];
+    <link rel="stylesheet" href="bower_components/bootstrap/dist/css/bootstrap.css">
+    <script type="text/javascript" src="bower_components/jquery/dist/jquery.js"></script>
+    <script type="text/javascript" src="clientes.js"></script>
+    <script type="text/javascript">
+        $(document).ready(function () {
+            var aux=10;
+            var aux1=0;
+            $("#search").click(function () {
+
+                for(var i=aux1;i<aux;i++) {
+                    $(clientes[i]).each(function (key, value) {
+
+                        if (value.activo - value.pasivo < 0) {
+                            var aux = "warning";
```

Comment:
@leamarty Si borras el each(), pero dejas esto como esta, el codigo te tira unos errores bizarrisimos porque estas redeclarando el scope de aux, lo que hace que cuando se evalue el for, aux sea undefined.
Sin embargo en este caso no tira ningun error, porque lo esta ejecutando dentro de each(), que tiene su propio scope.
Realmente un excelente ejemplo para aprender scopes, y cuando se pueden repetir nombres de variable sin problemas.